### DISCUSSION TITLE
Converting EPS files into WebP

### DISCUSSION THREAD

**Discussion by mamkarolina** [ImageMagick/ImageMagick#7594] [Help]
### ImageMagick version

7.1.1-11

### Operating system, version and so on

Windows10, SaaS

### Description

Hi dear community

I am looking for a well-functioning sequence of commands to transform eps files into webp's, where scaling (my example 1280x1280), positioning (centered) and orientation (as from original) are ensured. 

my two steps look like this, where in the first I use ghostscript and in the second one ImageMagick:
```
"ArgumentString": "-sDEVICE=pngalpha -dNOPAUSE -dBATCH -dQUIET -dSAFER -r300 -dGraphicsAlphaBits=4 -dUseCIEColor -dEPSCrop -sstdout=%stderr -sOutputFile=\"{StepOutputFile}\" \"{StepInputFile}\""

"ArgumentString": "-quiet \"{StepInputFile}[0]\" -resize \"1280x1280>\" -density 72x72 -quality 92 -auto-orient -define webp:lossless=true -colorspace sRGB \"{StepOutputFile}\""
``` 
In the example above, I have the positioning correct, the orientation as well, but for some files the scaling doesn't seem to work (see an example in the screenshot attached).
- Original Size:  277x142 pixel (48,81mm x 97,62mm)
- Desired Size:  1280x1280 pixel
![image](https://github.com/user-attachments/assets/56563117-4bfa-4fde-8cd9-511b05970abd)

Any idea, what could be missing in the commands above so that all files are scaled correctly into a webp?
Or does it have to do with the original file. If so, what information or prerequisite is required?

**Comment by snibgo** [comment]
For ImageMagick, what is your "ArgumentString"? It should be "magick", not "convert" or "magick convert".

> but for some files the scaling doesn't seem to work

This is vague. What is wrong?

An EPS file may be entirely vector, which means it doesn't have a natural dimension in pixels. But it may have physical dimensions, eg in inches. You can specify a density in pixels per inch, *before* the input file:
```
magick -quiet -density 300 in.eps -resize "1280x1280>" -density 72x72 -quality 92 -auto-orient -define webp:lossless=true -colorspace sRGB out.webp
```
It may help if you posted a sample input eps file (perhaps zip it and post the zip file), and explain what output you want.

**Comment by mamkarolina** [comment]
Hi @snibgo Thanks so much for your answer.
What I specifically mean with scaling doesn't work is: 
The webp file should have min. one side **1280** pixel. In my output, it only reached **1153** x 591
Find the original file, so as the webp in the ZIP attached.
[AW_IF_Awared_2024.zip](https://github.com/user-attachments/files/16895876/AW_IF_Awared_2024.zip)

**Comment by snibgo** [comment]
Your command includes:
```
-resize "1280x1280>"
```
See https://imagemagick.org/script/command-line-processing.php#geometry :
```
widthxheight> 	Shrinks an image with dimension(s) larger than the corresponding width and/or height argument(s).
```
So your command says: if the image is *larger* than that size, then reduce the size. But that isn't what you want. Instead of `>`, the flag you want is `^`.

**Comment by mamkarolina** [comment]
Thanks for the hint. 
Depending on the input image, I want to reduce the size _or_ enlarge the image to 1280x1280 without losing the aspect ratio and quality.

- Using ^ my output image reaches the correct size but loses quality.
- Using > the output image doesn't scale up (or down) and maintains quality
- Not using a flag also leads to an output image with correct size but loss of quality

I've tried using additional ghoscript steps to ensure quality, which does, but it leads to some files being rotated by 90° to the left (like the example image posted before, see screenshot). 

```
-dSAFER -dBATCH -dNOPAUSE -sDEVICE=pdfwrite -dEPSCrop -sOutputFile=
-dSAFER -dBATCH -dNOPAUSE -sDEVICE=eps2write -dPDFFitPage -sPAPERSIZE=letter -sOutputFile=\
``` 
![image](https://github.com/user-attachments/assets/79a4eb24-e75c-447c-afc5-4e390d61ddb2)

**Comment by snibgo** [comment]
When rasterizing a vector image, there is usually no need to enlarge the result, and the enlargement will be pixelated ("poor quality"). Instead, it is better to use a suitable `-density N` before reading the SVG. If necessary, reduce the size after rasterizing.

In some workflows, it may be necessary to rasterize twice. The first time will find the size (in pixels) with a certain density. That size is used to calculate the required density for a required size.

I don't know what causes a rotation. Please post an EPS that has that problem.

**Reply by mamkarolina** [reply]
Hi @snibgo I've found the solution for reaching a good quality while scaling up:
In the ghostscript step, as soon as I change the r-Parameter from r300 to r750, the output image maintains a good quality while scaled up.

```
"ArgumentString": "-sDEVICE=pngalpha -dNOPAUSE -dBATCH -dQUIET -dSAFER -r300 -dGraphicsAlphaBits=4 -dUseCIEColor -dEPSCrop -sstdout=%stderr -sOutputFile=\"{StepOutputFile}\" \"{StepInputFile}\""
```
